File: leetcode/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const trackerNums = {};
    
    for(let num of nums){
        if(trackerNums[num]){
            return true;
        }else{
            trackerNums[num] = true;
        }
    }
    
    
    return false;
};

File: leetcode/CPP/containsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
      unordered_set<int> tracker;
      
      for(int i = 0; i < nums.size(); i++){
        if(tracker.find(nums[i]) == tracker.end()){
          tracker.insert(nums[i]);
        }else{
          return true;
        }
      }
      
      return false;
    }
};
